code/game/objects/effects/spawners/minor_variation.dm

```diff
@@ -0,0 +1,58 @@
+/*
+Makes variations possible with code only.
+This is mainly to prevent creating new map files when creating smaller variations that aren't worth a map file. - Hopek
+*/
+
+/obj/effect/variation
+	name = "variation"
+	icon = 'icons/effects/landmarks_variation.dmi'
+	icon_state = "generic_green"
+	var/variations // number of variations
+	var/choice // chosen variation by choice()
+	var/chosen // has a choice been made for this place?
+	var/location_place // location of placement calculated from reference
+	var/ref_x // x of reference
+	var/ref_y // y of reference
+	var/ref_z // z of reference
+
+/obj/effect/variation/Initialize()
+	. = ..()
+	chosen = FALSE
+
+/obj/effect/variation/proc/place(x, y , item , direction)
+	if(!x)
+		x = 0
+	if(!y)
+		y = 0
+	if(item)
+		location_place = locate((ref_x + x) , (ref_y + y) , ref_z)
+		if(islist(item))
+			for(var/I in item)
+				if(direction)
+					var/obj/H = new I(location_place)
+					H.dir = direction
+					return
+				else
+					new I(location_place)
+		else
+			if(direction)
+				var/obj/H = new item(location_place)
+				H.dir = direction
+			else
+				new item(location_place)
+	return
+
+/obj/effect/variation/proc/choice()
+	if(choice == "" || choice == 0)
+		choice = 1
+	if(!chosen)
+		ref_x = src.x
```

Comment:
this is probably obvious in hindsight, but why src.x instead of x?
same for below

Reply:
It is best practice to include the object being acted upon for clarity's sake.
Essentially for readability's sake of making sure other coders that usually work in different languages can easily make the distinction

Reply:
Fair